Feature/reorganise arch structure

DrDeano opened the pull request:
Restructured the x86 ports to refactor common code

This adds a new folder structure structure:
- x86
  - 32bit
  - 64bit
  - common

Refactored arch, GDT, IDT and serial

SamTebbs33 commented on makeiso_64.sh:
```diff
@@ -23,7 +23,7 @@ then
 fi
 
-MAP_FILE="zig-cache/kernel.map"
+MAP_FILE="zig-out/kernel.map"
```
I get a zig-cache directory when building unlike the CI, which gets zig-out. It might be a good idea to pass `b.install_path` as an argument to the makeiso files so they can align to whatever path is being used.

DrDeano replied:
Sure I shall copy you bug fix over as well

SamTebbs33 reviewed (commented):
Very nice reorganisation :+1:

SamTebbs33 reviewed (approved):
Very nice

DrDeano committed:
Restructured the x86 ports to refactor common code

This adds a new folder structure structure:
- x86
  - 32bit
  - 64bit
  - common

Refactored arch, GDT, IDT and serial
Zig master update (9d409233b)

Updated to use install path for MAP file
Added bug fix #305
